src/Effect/Monad/Writer/Instances.agda

```diff
@@ -0,0 +1,26 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Instances for the writer monad
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Effect.Monad.Writer.Instances where
+
+open import Effect.Monad.Writer.Transformer
+
+instance
+  writerTFunctor = λ {s} {S} {f} {M} {{fun}} → functor {s} {S} {f} {M} fun
+  writerTApplicative = λ {s} {S} {f} {M} {{mo}} {{mon}} → applicative {s} {S} {f} {M} {mo} mon
+  writerTEmpty = λ {s} {S} {f} {M} {{e}} → empty {s} {S} {f} {M} e
+  writerTChoice = λ {s} {S} {f} {M} {{ch}} → choice {s} {S} {f} {M} ch
+  writerTApplicativeZero = λ {s} {S} {f} {M} {{mo}} {{mon}} → applicativeZero {s} {S} {f} {M} {mo} mon
+  writerTAlternative = λ {s} {S} {f} {M} {{mo}} {{mpl}} → alternative {s} {S} {f} {M} {mo} mpl
+  writerTMonad = λ {s} {S} {f} {M} {{mo}} {{mon}} → monad {s} {S} {f} {M} {mo} mon
+  writerTMonadZero = λ {s} {S} {f} {M} {{mo}} {{mz}} → monadZero {s} {S} {f} {M} {mo} mz
+  writerTMonadPlus = λ {s} {S} {f} {M} {{mo}} {{mpl}} → monadPlus {s} {S} {f} {M} {mo} mpl
+  writerTMonadT = λ {s} {S} {f} {M} {{mo}} {{mon}} → monadT {s} {S} {f} {M} {mo} mon
+  writerTMonadWriter = λ {s} {S} {f} {M} {{mon}} → monadWriter {s} {S} {f} {M} mon
+  writerTLiftReaderT = λ {s} {S₁} {S₂} {f} {M} {{fun}} {{mw}} → liftReaderT {s} {S₁} {S₂} {f} {M} fun mw
+  writerTLiftStateT = λ {s} {S₁} {S₂} {f} {M} {{fun}} {{mw}} → liftStateT {s} {S₁} {S₂} {f} {M} fun mw
```

Comment:
Do we need a lift for `ErrorT` too?

Reply:
It would be nice to have all lifts for all transformers but writing them by hand is a bitter experience.

What are the prospects of adding an `Effect.Algebraic` module or something? Seems like Agda's dependent types are much more accommodating to effect systems than Haskell which has a zoo of incompatible and inefficient ones.

Reply:
I think it's still an open research question. Edwin Brady had a paper a while ago
about doing something like that in Idris1 but he's come back from it for Idris2 and
I'm not sure we have a good story yet.